app/controllers/provider_interface/offer/conditions_controller.rb

```diff
@@ -0,0 +1,30 @@
+module ProviderInterface
+  module Offer
+    class ConditionsController < OffersController
+      def new
+        @wizard = OfferWizard.new(offer_store, { current_step: 'conditions' })
+        @wizard.save_state!
+      end
+
+      def create
+        @wizard = OfferWizard.new(offer_store, conditions_params.to_h)
+
+        if @wizard.valid_for_current_step?
+          @wizard.save_state!
+
+          redirect_to [:new, :provider_interface, @application_choice, :offer, @wizard.next_step]
```

Comment:
Is there a way to write this in a more readable way? It's not immediately clear to me where we are redirecting to (because of the array)

Reply:
The alternative was to use metaprogramming but I think that makes things even harder to undertand, whilst this format is not uncommon in rails.